### DISCUSSION TITLE
custom marker question

### DISCUSSION THREAD

**Discussion by farzanehDi** [xyflow/xyflow#3733] [Q&A]
When I added a custom icon, when the nodes are moved and the source handler is changed, the icon stays the same.
How can I handle icon rotation when moving?
![1](https://github.com/xyflow/xyflow/assets/50740353/dad8e395-8d15-476c-88f0-369e3b18c524)

![2](https://github.com/xyflow/xyflow/assets/50740353/bcc7932e-d847-4aa4-866c-3844ab4603b4)

codesandbox:
https://codesandbox.io/p/sandbox/peaceful-pond-q8ztfx

**Comment by moklick** [comment]
Hey @farzanehDi 

could you provide a codesandbox for this? You can use this as a starting point https://new.reactflow.dev

**Reply by farzanehDi** [reply]
hey @moklick 
I did it, you can see it now in the link below:
https://codesandbox.io/p/sandbox/peaceful-pond-q8ztfx

**Comment by farzanehDi** [comment]
Hi
does anyone have any ideas or solutions?

**Reply by Everduin94** [reply]
I'm not super experienced with xyflow. But, I believe the issue stems from a few things:

When you use `connection=loose` and set all handles to `source`. You need to apply a unique ID to each handle. Otherwise, xyflow behaves inconsistently / unexpectedly

```jsx
const CustomNode = ({ data, id }: NodeProps) => {
  return (
    <>
      <Handle id={`${id}-L`} type="source" position={Position.Left} />
      {id}
      <Handle id={`${id}-R`} type="source" position={Position.Right} />
    </>
  );
};
```

Then, in your data, you need to define the sourceHandle and targetHandle.

```ts
const initialEdges: Edge[] = [
  {
    id: "e1-3",
    type: "customEdge",
    source: `5`,
    target: `4`,
    markerEnd: "customMarker",
    sourceHandle: `5-R`,
    targetHandle: `4-L`,
  },
];
```

Hope this helps!
